core/src/nodes/raw_swarm.rs

```diff
@@ -760,19 +788,26 @@ where
         <THandler::Handler as NodeHandler>::OutboundOpenInfo: Send + 'static, // TODO: shouldn't be necessary
         THandlerErr: error::Error + Send + 'static,
     {
-        // Start by polling the listeners for events.
+        // Start by polling the listeners for events, but only
+        // if numer of incoming connection does not exceed the limit.
         match self.listeners.poll() {
```

Comment:
I think we should simply not poll the listeners if we are full. In other words, put an `if` around the `match self.listeners.poll()`.